### PR TITLE
.NET8 upgrade for batch 2 of dfc-coursedirectory repo

### PR DESCRIPTION
**Pull request by mimran-DFE**
.NET8 upgrade for batch 2 of dfc-coursedirectory repo. This batch includes following projects:

- Dfc.CourseDirectory.Api
- Dfc.CourseDirectory.Api.Tests
- Dfc.CourseDirectory.Functions
- Dfc.CourseDirectory.Functions.Tests
- Dfc.CourseDirectory.FindACourseApi
- Dfc.CourseDirectory.FindACourseApi.Tests

### CODE REVIEW

File: src/Dfc.CourseDirectory.Functions/FunctionInstanceServicesCatalog.cs [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
+        _instanceServices.TryGetValue(instanceId, out var serviceProvider);
+        return serviceProvider;
+    }
 
-        T IJobActivatorEx.CreateInstance<T>(IFunctionInstanceEx functionInstance)
-        {
-            _instanceServices.Add(functionInstance, functionInstance.InstanceServices);
-            return _innerActivator.CreateInstance<T>(functionInstance);
-        }
+    public void AddFunctionInstance(Guid instanceId, IServiceProvider serviceProvider)
```
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.Functions/FunctionInstanceServicesCatalog.cs:18]: What is the purpose of the addfunctioninstance and removefunction instance methods here? They didnt exist before the upgrade and dont seem to be referenced by anything 
image

File: src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs:1]: Lots of commented out code here. Are these going to be resolved in a later batch ticket or are they now redundant?

File: src/Dfc.CourseDirectory.Web/Controllers/EditCourse/EditCourseRunController.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.Web/Controllers/EditCourse/EditCourseRunController.cs:1]: Same here with commented out code

File: src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/ChooseQualification.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/ChooseQualification.cs:1]: More commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/CourseRun.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/CourseRun.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/Errors.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/Errors.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/ResolveRowDetails.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/ResolveRowDetails.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/Details.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/Details.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/SelectTLevel.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/SelectTLevel.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/ViewAndEditTLevel/EditTLevel.cs
Comment (ajones-DFE) [inline src/Dfc.CourseDirectory.WebV2/Features/TLevels/ViewAndEditTLevel/EditTLevel.cs:1]: Commented out rules

File: tests/Dfc.CourseDirectory.WebV2.Tests/ValidationTests/DateValidationTests.cs
Comment (ajones-DFE) [inline tests/Dfc.CourseDirectory.WebV2.Tests/ValidationTests/DateValidationTests.cs:1]: Commented out rules

File: src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs:1]: Commented out rules will be fixed in next batch. Currently working on them.

File: src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/ChooseQualification.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/ChooseQualification.cs:1]: Commented out rules will be fixed in next batch. Currently working on them.

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/ResolveRowDetails.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/ResolveRowDetails.cs:1]: Commented out rules will be fixed in next batch as these rules are in Dfc.CourseDirectory.Web projects. Currently working on them.

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/SelectTLevel.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.WebV2/Features/TLevels/AddTLevel/SelectTLevel.cs:1]: Commented out rules will be fixed in next batch as these rules are in Dfc.CourseDirectory.Web projects. Currently working on them.

File: src/Dfc.CourseDirectory.WebV2/Features/TLevels/ViewAndEditTLevel/EditTLevel.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.WebV2/Features/TLevels/ViewAndEditTLevel/EditTLevel.cs:1]: Commented out rules will be fixed in next batch as these rules are in Dfc.CourseDirectory.Web projects. Currently working on them.

File: tests/Dfc.CourseDirectory.WebV2.Tests/ValidationTests/DateValidationTests.cs
Comment (mimran-DFE) [reply tests/Dfc.CourseDirectory.WebV2.Tests/ValidationTests/DateValidationTests.cs:1]: Commented out rules will be fixed in next batch as these rules are in Dfc.CourseDirectory.Web projects. Currently working on them.

File: src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/CourseRun.cs
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.WebV2/Features/ChooseQualification/CourseRun.cs:1]: Commented out rules will be fixed in next batch as these rules are in Dfc.CourseDirectory.Web projects. Currently working on them.

File: src/Dfc.CourseDirectory.Functions/FunctionInstanceServicesCatalog.cs (hunk 1 above)
Comment (mimran-DFE) [reply src/Dfc.CourseDirectory.Functions/FunctionInstanceServicesCatalog.cs]: Yes you are right. They don't seem to be required here. I have removed these methods now

**Review by ajones-DFE — APPROVED** [review]

**Pull request merged**